dashboard/app/models/circuit_playground_discount_code.rb

```diff
@@ -20,11 +20,21 @@ def self.claim(full_discount)
 
     code = nil
     Retryable.retryable(on: ActiveRecord::RecordNotSaved) do
-      code = where(full_discount: full_discount).
+      codes = where(full_discount: full_discount).
         where(claimed_at: nil).
         where(voided_at: nil).
         where(expiration_field.gt(Time.now)).
-        first
+        limit(20)
+
+      if codes.length < 20
```

Comment:
Are you sure `length` is the right call here?  It [looks like `size` might be a better choice](https://mensfeld.pl/2014/09/activerecord-count-vs-length-vs-size-and-what-will-happen-if-you-use-it-the-way-you-shouldnt/).